src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/KeyboardEventSender.cs

```diff
@@ -83,6 +94,34 @@ private static void QueueTextEvent(int replaySegment, Key key)
                 RGDebug.LogInfo($"({replaySegment}) Sending Text Event for char: '{value}'");
                 InputSystem.QueueEvent(ref inputEvent);
             }
+            
+            // If there are active UI input fields, simulate a KeyDown UI event for newly pressed keys
```

Comment:
I feel like this code should be wrapped in LEGACY # blocks as I don't think it is necessary for the newer input system.  If a field is selected in the UI, then text events will go to that active field.

Reply:
I wasn't able to record/replay input field entry (e.g. the profile name in Boss Room) until I added this code. I'm not sure that these TextEvents actually make it to the InputFields, which are reading keyboard events via `Event.PopEvent` from the UnityGUI event system?

Reply:
interesting.. it indeed does not work

tested by recording bossroom and replaying;  last time i tried this on PW it WAS working, but that was forwarding the text event to gameface and re-rendering.  That should be impacted by this new code as it is not a unity UI component

Reply:
I guess the only part that feels weird is `RGLegacyInputUtils` methods for the new input system as the event system for UI components

Reply:
I've moved the UI key event creation/sending methods out of RGLegacyInputUtils and into KeyboardEventSender. This seems like the most appropriate place for them since they are specific to sending keyboard events and aren't related to either the legacy Input Manager or new Input System (they are part of the UnityGUI's own event system).